test/Microsoft.Extensions.Logging.Analyzer.Test/FormatStringAnalyzerTests.cs

```diff
@@ -12,60 +13,90 @@ namespace Microsoft.Extensions.Logging.Analyzer.Test
     public class FormatStringAnalyzerTests : DiagnosticVerifier
     {
         [Theory]
-        [MemberData(nameof(GenerateTemplateUsages), @"""{0}"", 1")]
-        public void DiagnosticIsProducedForNumericFormatArgument(string format)
+        [MemberData(nameof(GenerateTemplateUsages), @"""{0}""", "1", true)]
+        public void MEL0001IsProducedForNumericFormatArgument(string format)
         {
-            var diagnostic = Assert.Single(GetDiagnostics(format));
+            // Enable MEL0005 because it shouldn't trigger on numeric arguments and we want to verify that.
+            var diagnostic = Assert.Single(GetDiagnostics(format, "MEL0005"));
             Assert.Equal("MEL0001", diagnostic.Id);
         }
 
         [Theory]
-        [MemberData(nameof(GenerateTemplateUsages), @"""{camelCase}"", 1")]
-        public void DiagnosticIsProducedForCamelCasedFormatArgument(string format)
+        [MemberData(nameof(GenerateTemplateUsages), @"$""{string.Empty}""", "", true)]
+        [MemberData(nameof(GenerateTemplateUsages), @"""string"" + 2", "", true)]
+        public void MEL0002IsProducedForDynamicFormatArgument(string format)
         {
-            var diagnostic = Assert.Single(GetDiagnostics(format, "MEL0005"));
-            Assert.Equal("MEL0005", diagnostic.Id);
+            var diagnostic = Assert.Single(GetDiagnostics(format));
+            Assert.Equal("MEL0002", diagnostic.Id);
         }
 
         [Theory]
-        [MemberData(nameof(GenerateTemplateUsages), @"$""{string.Empty}""")]
-        [MemberData(nameof(GenerateTemplateUsages), @"""string"" + 2")]
-        public void DiagnosticIsProducedForDynamicFormatArgument(string format)
+        [MemberData(nameof(GenerateTemplateUsages), @"""{string}""", "1, 2", false)]
+        [MemberData(nameof(GenerateTemplateUsages), @"""{str"" + ""ing}""", "1, 2", false)]
+        [MemberData(nameof(GenerateTemplateUsages), @"""{"" + nameof(ILogger) + ""}""", "", false)]
+        [MemberData(nameof(GenerateTemplateUsages), @"""{"" + Const + ""}""", "", false)]
```

Comment:
Why false here?

Reply:
The next test is verifying Define cases, since they involve a different pattern (generating generic args instead of specifying expressions). It seemed cleaner.